### PR TITLE
Fix compilation on OSX 10.9

### PR DESCRIPTION
**Pull request by iamphi**
Found solution from: 
https://github.com/eclipse/omr/issues/902#issuecomment-288559044

### CODE REVIEW

File: mono/utils/mono-state.c [hunk 1]
```diff
@@ -33,6 +33,17 @@ extern GCStats mono_gc_stats;
 #ifdef TARGET_OSX
 #include <mach/mach.h>
 #include <mach/task_info.h>
+
+// OSX 10.9 does not have MAP_ANONYMOUS
+#if !defined(
+  #define NO_MAP_ANONYMOUS
```
Comment (vargaz) [inline mono/utils/mono-state.c]: This doesn't look right.

Comment (iamphi) [reply mono/utils/mono-state.c]: Oops. Copy+paste error. I did test this change in my custom mono compilation on OSX 10.9

File: mono/utils/mono-state.c [hunk 2]
```diff
@@ -33,6 +33,17 @@ extern GCStats mono_gc_stats;
 #ifdef TARGET_OSX
 #include <mach/mach.h>
 #include <mach/task_info.h>
+
+// OSX 10.9 does not have MAP_ANONYMOUS
+#if !defined(MAP_ANONYMOUS)
+  #define NO_MAP_ANONYMOUS
+  #if defined(MAP_ANON)
+    #define MAP_ANONYMOUS MAP_ANON
```
Comment (jaykrell) [inline mono/utils/mono-state.c]: Does this need to be conditional or is MAP_ANON actually always available/defined?

**Comment by marek-safar** [comment]
approve

**Comment by lewing** [comment]
@monojenkins build Project Files Diff

**Comment by lewing** [comment]
@monojenkins build Windows x64 FullAOT+LLVM

**Pull request merged**